crates/component-macro/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+use proc_macro2::{Literal, TokenStream, TokenTree};
+use quote::quote;
+use syn::{parse_macro_input, parse_quote, Data, DeriveInput, Error, Result};
+
+static INCOMPATIBLE_TYPE_MESSAGE: &str =
+    "a `record` must be defined as a `struct` with at least one named field";
```

Comment:
Personally I think it's ok to inline all these at the use-site, otherwise I find myself tabbing back and forth to what exactly this error message is saying